pkg/config/validate.go

```diff
@@ -109,20 +110,22 @@ func (dc DeploymentConfig) executeValidators() error {
 	}
 
 	if warned || errored {
-		log.Println("validator failures can indicate a credentials problem.")
-		log.Println("cloud credentials associated with your Google Cloud account can be generated by the following command:")
-		log.Println("  $ gcloud auth application-default login")
```

Comment:
One side effect of this change is that the recommendation to call `gcloud auth application-default login` is getting buried/dropped. 

1. Do we think this is still a valid reason for failure (even if it is one of several)?
2. Do we print anything about `gcloud auth` in the messages referenced by "See messages above for suggested"?
3. Should we add what a gcloud auth failure looks like in troubleshooting doc?

Reply:
Sorry, I meant to point this out explicitly in the description.

 (1) = yes, (2) = yes, if you count a link to those instructions. We handle it better now.

- Recent changes address [the case when there are no loadable application default credentials](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/6526126fe1fa6bc4972c7c754e8aee52ee799de6/pkg/validators/validators.go#L41-L48)
  - Every `test_x_y` validator trips over this code 

This handles the "user has no application default credentials" case which that message addresses.

(3)

Otherwise, it is hard and maybe impossible to disambiguate "I have the wrong credentials"  from "I have the right credentials but my boss won't give me the right IAM role from "I'm logged in under my personal gmail account but forgot". I believe all cases return a simple 404 Not Found because you don't want unauthorized users to detect the presence of projects by just poking around. (I am willing to believe that I may have something wrong here, though)

My instinct is that these cases can't be addressed in code but in improved documentation guiding the user around the difference between `gcloud auth login` and "application default credentials" and how to know which you're using. I'm a bit hesitant to start writing things without a bit of real world experience of the failures users have. We should rely on cloud.google.com as much as possible.

Reply:
I am happy with (2) in place.